cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/preview/DefaultPreviewRunner.java

```diff
@@ -212,11 +229,15 @@ protected void startUp() throws Exception {
 
   @Override
   protected void shutDown() throws Exception {
+    shutDownUnrequiredServices();
+    datasetService.stopAndWait();
+  }
+
+  private void shutDownUnrequiredServices() {
```

Comment:
comment is not valid, the listener is on program lifecycle service. i thought it was preview server.